src/main/java/digital/slovensko/autogram/core/SigningJob.java

```diff
@@ -95,43 +102,41 @@ public String getDocumentAsPlainText() {
         }
     }
 
-    private String transform() {
+    private String transform()
+            throws IOException, ParserConfigurationException, SAXException, TransformerException {
         // TODO probably move this logic into signing job creation
-        try (var is = this.document.openStream()){
-            var builderFactory = DocumentBuilderFactory.newInstance();
-            builderFactory.setNamespaceAware(true);
-
-            var inputSource = new InputSource(is);
-            inputSource.setEncoding(encoding.displayName());
-            var document = builderFactory.newDocumentBuilder().parse(inputSource);
-
-            var xmlSource = new DOMSource(document);
-            if (isXDC())
-                xmlSource = extractFromXDC(document, builderFactory);
-
-            var outputTarget = new StreamResult(new StringWriter());
-            var transformer = TransformerFactory
-                    .newDefaultInstance().newTransformer(
-                            new StreamSource(new ByteArrayInputStream(parameters.getTransformation().getBytes(encoding)))
-                    );
-            var outputProperties = new Properties();
-            outputProperties.setProperty(OutputKeys.ENCODING, encoding.displayName());
-            transformer.setOutputProperties(outputProperties);
-            transformer.transform(xmlSource, outputTarget);
-
-            return outputTarget.getWriter().toString().trim();
-        } catch (Exception e) {
-            e.printStackTrace();
-            return null; // TODO
-        }
+        var is = this.document.openStream();
```

Comment:
Tento `try` tu bol kvoli tomu, aby to ten stream potom zavrelo.

Reply:
hm, myslim, ze som ho odstranoval kvoli tomu, ze som tu dostaval "self-suppression not permitted" error, ale idem to vyskusat

Reply:
```
Starting with args: []
java.io.UTFDataFormatException: encoded string (iVBORw0K...rkJggg==) too long: 153768 bytes
        at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:369)
        at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:333)
        at java.xml/com.sun.org.apache.bcel.internal.classfile.ConstantUtf8.dump(ConstantUtf8.java:182)
        at java.xml/com.sun.org.apache.bcel.internal.classfile.ConstantPool.dump(ConstantPool.java:237)
        at java.xml/com.sun.org.apache.bcel.internal.classfile.JavaClass.dump(JavaClass.java:306)
        at java.xml/com.sun.org.apache.bcel.internal.classfile.JavaClass.dump(JavaClass.java:292)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.compiler.XSLTC.dumpClass(XSLTC.java:905)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.compiler.Stylesheet.translate(Stylesheet.java:740)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.compiler.XSLTC.compile(XSLTC.java:505)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.compiler.XSLTC.compile(XSLTC.java:575)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl.newTemplates(TransformerFactoryImpl.java:1036)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl.newTransformer(TransformerFactoryImpl.java:834)
        at digital.slovensko.autogram.core.SigningJob.transform(SigningJob.java:124)
        at digital.slovensko.autogram.core.SigningJob.getDocumentAsHTML(SigningJob.java:156)
        at digital.slovensko.autogram.ui.gui.SigningDialogController.lambda$4(SigningDialogController.java:148)
        at javafx.base/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:181)
        at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
        at javafx.base/javafx.beans.property.ReadOnlyObjectPropertyBase.fireValueChangedEvent(ReadOnlyObjectPropertyBase.java:80)
        at javafx.base/javafx.beans.property.ReadOnlyObjectWrapper.fireValueChangedEvent(ReadOnlyObjectWrapper.java:102)
        at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
        at javafx.base/javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
        at javafx.web/javafx.scene.web.WebEngine$LoadWorker.updateState(WebEngine.java:1264)
        at javafx.web/javafx.scene.web.WebEngine$LoadWorker.dispatchLoadEvent(WebEngine.java:1379)
        at javafx.web/javafx.scene.web.WebEngine$PageLoadListener.dispatchLoadEvent(WebEngine.java:1244)
        at javafx.web/com.sun.webkit.WebPage.fireLoadEvent(WebPage.java:2543)
        at javafx.web/com.sun.webkit.WebPage.fwkFireLoadEvent(WebPage.java:2388)
        at javafx.web/com.sun.webkit.network.URLLoaderBase.twkDidFinishLoading(Native Method)
        at javafx.web/com.sun.webkit.network.URLLoader.notifyDidFinishLoading(URLLoader.java:776)
        at javafx.web/com.sun.webkit.network.URLLoader.lambda$didFinishLoading$5(URLLoader.java:767)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
Exception in thread "JavaFX Application Thread" java.lang.IllegalArgumentException: Self-suppression not permitted
        at java.base/java.lang.Throwable.addSuppressed(Throwable.java:1072)
        at digital.slovensko.autogram.core.SigningJob.transform(SigningJob.java:132)
        at digital.slovensko.autogram.core.SigningJob.getDocumentAsHTML(SigningJob.java:156)
        at digital.slovensko.autogram.ui.gui.SigningDialogController.lambda$4(SigningDialogController.java:148)
        at javafx.base/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:181)
        at javafx.base/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
        at javafx.base/javafx.beans.property.ReadOnlyObjectPropertyBase.fireValueChangedEvent(ReadOnlyObjectPropertyBase.java:80)
        at javafx.base/javafx.beans.property.ReadOnlyObjectWrapper.fireValueChangedEvent(ReadOnlyObjectWrapper.java:102)
        at javafx.base/javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
        at javafx.base/javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
        at javafx.web/javafx.scene.web.WebEngine$LoadWorker.updateState(WebEngine.java:1264)
        at javafx.web/javafx.scene.web.WebEngine$LoadWorker.dispatchLoadEvent(WebEngine.java:1379)
        at javafx.web/javafx.scene.web.WebEngine$PageLoadListener.dispatchLoadEvent(WebEngine.java:1244)
        at javafx.web/com.sun.webkit.WebPage.fireLoadEvent(WebPage.java:2543)
        at javafx.web/com.sun.webkit.WebPage.fwkFireLoadEvent(WebPage.java:2388)
        at javafx.web/com.sun.webkit.network.URLLoaderBase.twkDidFinishLoading(Native Method)
        at javafx.web/com.sun.webkit.network.URLLoader.notifyDidFinishLoading(URLLoader.java:776)
        at javafx.web/com.sun.webkit.network.URLLoader.lambda$didFinishLoading$5(URLLoader.java:767)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
Caused by: javax.xml.transform.TransformerConfigurationException: This Templates does not contain a class with the name 'die.verwandlung.GregorSamsa'.
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TemplatesImpl.defineTransletClasses(TemplatesImpl.java:531)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TemplatesImpl.getTransletInstance(TemplatesImpl.java:554)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TemplatesImpl.newTransformer(TemplatesImpl.java:587)
        at java.xml/com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl.newTransformer(TransformerFactoryImpl.java:835)
        at digital.slovensko.autogram.core.SigningJob.transform(SigningJob.java:124)
        ... 20 more
```

Reply:
Sry ze sa sem miesam, ale dostavam notifikacie a toto ma zaujalo :) Ako vyzera ten kod, ked dostavas tu exception? Ten stacktrace je zvlastny, zaroven ak si pozries java.lang.Throwable#addSuppressed implementaciu, tak ta chyba nastava len pre pripad
```
if (exception == this)
            throw new IllegalArgumentException(SELF_SUPPRESSION_MESSAGE, exception);
```
co teda netusim ako nastava.

Reply:
vpohode miesaj sa :) 
deje sa to ked to 
- je tento catch, tak ako teraz v maine https://github.com/slovensko-digital/autogram/blob/93e22e18c02dc578c5f64a9e3099063517d1a6eb/src/main/java/digital/slovensko/autogram/core/SigningJob.java#L100
- transformer sa konstruuje cez defaultnu instanciu (nie saxon) https://github.com/slovensko-digital/autogram/blob/93e22e18c02dc578c5f64a9e3099063517d1a6eb/src/main/java/digital/slovensko/autogram/core/SigningJob.java#LL113C3-L113C3
- pride na vstup transformeru XML ktore ma v property prilis velky string (bas64 obrazok)

co znamena ze SigningJob.transform() throwne co vsak bezi v ui threade takze tipujem ze sa tam zamiesa nejako JavaFX  

a ked sa tak pocuvam tak by riesenim mohlo byt robit transformaciu na nejakom work threade (co by sme mali aj tak robit)

Reply:
Viem to nasimulovat, ale nedeje sa tam nic co by mi objasnilo co sa deje. 
<img width="293" alt="image" src="https://github.com/slovensko-digital/autogram/assets/1194439/a2c876fc-063f-4d04-a8f3-142f96586e5d">
<img width="794" alt="image" src="https://github.com/slovensko-digital/autogram/assets/1194439/713901d3-b56c-4d9f-9792-4c5d8099c036">

A ked tam dam breakpoint a step-debugujem tak sa self-suppression neudeje 😃

Reply:
commitol som tie dve verzie aby si to vedel pozriet

transform_try_with_resources
https://github.com/slovensko-digital/autogram/pull/207/files#diff-809281c5e5c91a48ef59ab424fe30afad2f49ad30ba50a46c3b3dedad13ecadcR110

transform_try_catch
https://github.com/slovensko-digital/autogram/pull/207/files#diff-809281c5e5c91a48ef59ab424fe30afad2f49ad30ba50a46c3b3dedad13ecadcR142

Reply:
na to aby to hodilo self-suppression
- pouzi `docs/transformation.html` na testovanie (aby sa cez api poslala aj transformacia ktora je velka)
- musi byt pouzity `var transformerFactory = TransformerFactory.newDefaultInstance();`
- musi byt pouzity transform_try_with_resources

(ako IDE mam vscode takze moze to byt aj bug v ich debuggeri)